src/plugins/elasticsearch/index.js

```diff
@@ -59,7 +59,9 @@ module.exports = function (kibana) {
       );
 
       // Set up the health check service and start it.
-      healthCheck(this, server).start();
+      var hc = healthCheck(this, server);
+      server.expose('waitUntilReady', hc.waitUntilReady);
```

Comment:
`hc.waitUntilReady` doesn't rely on `this` (nothing in health_check does) because it doesn't track its own state. The only state is tracked on the plugin itself, which is explicitly passed to healthcheck on the line prior.

Moving it to an arrow function wouldn't make it maintain it's context either, it'd just make it invoked within the context of elasticsearch/index instead of elasticsearch/health_check.

To guarantee that it is invoked within its own context, we'd need to explicitly bind it as a new function: `hc.waitUntilReady.bind(hc)`, but as I said before, that wouldn't actually provide us any benefit.